src/main/java/appeng/core/features/registries/grinder/GrinderRecipeManager.java

```diff
@@ -393,4 +393,117 @@ else if( item != other.item )
 		}
 
 	}
+
+	@Override
+	public IGrinderRecipeBuilder builder()
+	{
+		return new Builder();
+	}
+
+	/**
+	 * Internal {@link IInscriberRecipeBuilder} implementation.
+	 * Needs to be adapted to represent a correct {@link IInscriberRecipe}
+	 */
+	private static final class Builder implements IGrinderRecipeBuilder
+	{
+
+		private ItemStack in;
+		private ItemStack out;
+
+		private float optionalChance;
+		private ItemStack optionalOutput;
+
+		private float optionalChance2;
+		private ItemStack optionalOutput2;
+
+		private int turns;
+
+		@Override
+		public IGrinderRecipeBuilder withInput( ItemStack input )
+		{
+			this.in = this.copy( input );
+
+			return this;
+		}
+
+		@Override
+		public IGrinderRecipeBuilder withOutput( ItemStack output )
+		{
+			this.out = this.copy( output );
+
+			return this;
+		}
+
+		@Override
+		public IGrinderRecipeBuilder withFirstOptional( ItemStack optional, float chance )
+		{
+			this.optionalOutput = this.copy( optional );
+			this.optionalChance = chance;
+
+			return this;
+		}
+
+		@Override
+		public IGrinderRecipeBuilder withSecondOptional( ItemStack optional, float chance )
+		{
+			this.optionalOutput2 = this.copy( optional );
+			this.optionalChance2 = chance;
+
+			return this;
+		}
+
+		@Override
+		public IGrinderRecipeBuilder withTurns( int turns )
+		{
+			this.turns = turns;
+
+			return this;
+		}
+
+		@Nonnull
+		@Override
+		public IGrinderRecipe build()
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) The Cyclomatic Complexity of this method "build" is 18 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)